examples/plugin/main.go

```diff
@@ -94,6 +95,7 @@ func run() error {
 
 	pluginExecutor, err := plugin.NewTreeWalkPluginExecutor(treeWalker, []core.Plugin{
 		&exampleTreePlugin{},
+		&python_dependency_usage.DependencyUsagePlugin{},
```

Comment:
This plugin should have a constructor, something like `NewDependencyUsagePlugin(...)` and it should allow the caller to pass a callback function so that the caller can receive output from the plugin.

Imagine I want to build an app that uses this plugin to identify dependency usage and store them in a JSON file. I would want this plugin to somehow `callback` to my app code so that I can do what I need with the information

Reply:
i've now integrated a callback for streaming events back to caller